clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -1051,9 +1051,9 @@ private void handleApiVersionsResponse(List<ClientResponse> responses,
             apiVersionsResponse.data().finalizedFeaturesEpoch());
         apiVersions.update(node, nodeVersionInfo);
         this.connectionStates.ready(node);
-        log.debug("Node {} has finalized features epoch: {}, finalized features: {}, supported features: {}, ZK migration ready: {}, API versions: {}.",
+        log.debug("Node {} has finalized features epoch: {}, finalized features: {}, supported features: {}, API versions: {}.",
                 node, apiVersionsResponse.data().finalizedFeaturesEpoch(), apiVersionsResponse.data().finalizedFeatures(),
-                apiVersionsResponse.data().supportedFeatures(), apiVersionsResponse.data().zkMigrationReady(), nodeVersionInfo);
```

Comment:
(Unrelated to this PR) @mumrah Do we have a plan to remove this field from the protocol api? We would need to ensure it's done in a compatible way, but it would be nice not to have this field in request logs and general logging.

Reply:
Since it's a tagged field and 4.0 cannot be the target of a ZK migration, we could just remove it without a version bump. However, we might consider keeping it around so that 4.0+ controllers can reject errant registrations from a Zk broker (like due to a misconfiguration).

Reply:
Interesting, would this be from a broker that is being migrated?

Reply:
Oh actually I was confusing this with the same field in the broker registration request. Looking at usages in 3.9, I'm not sure we ever used this field. I'll dig in a bit on this.

Reply:
This field (ZkMigrationReady in ApiVersionsResponse) was last used in 3.6. Starting in 3.7, we used the controller registration RPC in lieu of this field. So, it seems a bit like dead code at this point. We should probably leave the field in the RPC so we don't reuse the tagged field number, but we can set the max versions to v4 and remove the usages in the non-generated code.

Reply:
Sounds good.

Reply:
> we can set the max versions to v4 

Pardon me, it seems to me the max version is already set to v4 (`ApiKeys.API_VERSIONS.latestVersion()`). Please correct me if I misunderstand anything.

> we might consider keeping it around so that 4.0+ controllers can reject errant registrations from a Zk broker (like due to a misconfiguration).

it seems that is addressed already.

https://github.com/apache/kafka/blob/be7415cb8bf5e09ad969cccfcbffd30fd68804ea/metadata/src/main/java/org/apache/kafka/controller/ClusterControlManager.java#L362

> remove the usages in the non-generated code.

I prefer to remove `isMigratingZkBroker` and `IsMigratingZkBroker` from non-generated code. similar to my previous comment https://github.com/apache/kafka/pull/17293#discussion_r1781513673